specs/network_details_for_hardware_inventory.adoc

```diff
@@ -0,0 +1,150 @@
+= Specification for adding network details in node hardware inventory
+
+Add network details also as a part of node hardware inventory
+
+== Problem description
+
+Hardware inventory details are used to show exact status of the particular
+node. Currently network details are missing in node hardware inventory.
+we have to add network details like network_id, ipv4, ipv6, netmask, status,
+interface_id, sysfs_id, device_link, interface_type, model, driver_modules, drive
+hw_address, link_detected as a part of node hardware inventory.
+
+== Use Cases
+
+Tendrl should show the current status of network interface.
+
+== Proposed change
+* From node agent collect all network interface details in node inventory list and
+  store it in etcd. (Detailed information in implementation section).
+
+== Alternatives
+
+* We can also use ethtool for getting network details. But ethool is not
```

Comment:
Please be aware, that ethtool provide very limited information if launched on virtual machine for `virtio` type of NIC. I even think, that some information are not available at all for `virtio` type of NIC (for example link `Speed`).

I know, that this didn't looks like big problem for production environment (where probably all storage nodes will be physical machines), but it might be problem for testing, examination and demo purposes.